docs/selecting-a-datastore.md

```diff
@@ -0,0 +1,114 @@
+# Selecting a datastore
+
+SpiceDB ships with a number of [datastores](/internal/datastore), drivers used to store and retrieve the relationship data used for computing permissions.
+
+The currently available datastores are:
+- [Cockroach DB](#cockroach-db) - Best for large or scalable installations
+- [Postgres](#postgres) - Best for smaller or isolated applications
+- [memdb](#memdb) - For development and testing
+
+## Cockroach DB
+
+[Implementation](/internal/datastore/crdb)
+
+Datastore for storing and accessing data in a [Cockroach DB](https://github.com/cockroachdb/cockroach) cluster.
+
+- Best for large or scalable installations
+- Fully distributed and resilient to node failures
+- Queries are sharded over the breadth of the cluster
+
+Caveats:
+
+- Setup and maintanence complexity of the Cockroach DB cluster
+
+### Running with CockroachDB datastore
+
+```sh
+--datastore-kind=cockroachdb --datastore-conn-uri="connection uri here"
+```
+
+Optional Parameters:
+
+| Parameter | Description | Example |
+| --------- | ----------- | ------- |
+| `datastore-max-tx-retries` | Maximum number of times to retry a query before raising an error | `--datastore-max-tx-retries=50` |
+| `datastore-tx-overlap-strategy` | The overlap strategy to prevent New Enemy on CRDB (see below) | `--datastore-tx-overlap-strategy=static` |
+| `datastore-tx-overlap-key` | The key to use for the overlap strategy (see below) | `--datastore-tx-overlap-key="foo"` |
+| `datastore-conn-max-idletime` | Maximum idle time for a connection before it is recycled | `--datastore-conn-max-idletime=60s` |
+| `datastore-conn-max-lifetime` | Maximum lifetime for a connection before it is recycled | `--datastore-conn-max-lifetime=300s` |
+| `datastore-conn-max-open` | Maximum number of concurrent connections to open | `--datastore-conn-max-open=10` |
+| `datastore-conn-min-open` | Minimum number of concurrent connections to open | `--datastore-conn-min-open=1` |
+| `datastore-query-split-size` | The (estimated) query size at which to split a query into multiple queries | `--datastore-query-split-size=5kb` |
+| `datastore-gc-window` | Sets the window outside of which overwritten relationships are no longer accessible | `--datastore-gc-window=1s` |
+| `datastore-revision-fuzzing-duration` | Sets a fuzzing window on all zookies/zedtokens | `--datastore-revision-fuzzing-duration=50ms` |
+| `datastore-readonly` | Places the datastore into readonly mode | `--datastore-readonly=true` |
```

Comment:
Can we format these tables so that they are legible in plaintext? You can use http://markdowntable.com if your editor doesn't have a plugin/macro for it.

Reply:
Sure